app/src/main/kotlin/at/bitfire/davdroid/ui/DebugInfoActivity.kt

```diff
@@ -120,101 +166,216 @@ class DebugInfoActivity : AppCompatActivity() {
+                    }
+                }
+            }
+            (localResource to remoteResource)
+                .takeIf { (l, r) -> l != null || r != null }
+                ?.let { (local, remote) ->
+                    item {
+                        CardWithImage(
+                            title = stringResource(R.string.debug_info_involved_caption),
+                            subtitle = stringResource(R.string.debug_info_involved_subtitle),
+                            icon = Icons.Rounded.Adb,
+                            modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp)
+                        ) {
+                            if (remote != null) {
+                                Text(
+                                    text = stringResource(R.string.debug_info_involved_remote),
+                                    style = MaterialTheme.typography.body1
+                                )
+                                Text(
+                                    text = remote,
+                                    fontFamily = FontFamily.Monospace
+                                )
+                            }
+                            if (local != null) {
+                                Text(
+                                    text = stringResource(R.string.debug_info_involved_local),
+                                    style = MaterialTheme.typography.body1
+                                )
+                                Text(
+                                    text = local,
+                                    fontFamily = FontFamily.Monospace
+                                )
+                            }
+                        }
+                    }
+                }
+            logFile?.let { logs ->
+                item {
+                    CardWithImage(
+                        title = stringResource(R.string.debug_info_logs_caption),
+                        subtitle = stringResource(R.string.debug_info_logs_subtitle),
+                        icon = Icons.Rounded.BugReport,
+                        modifier = Modifier.padding(horizontal = 8.dp, vertical = 8.dp)
+                    ) {
+                        TextButton(
+                            onClick = { shareFile(logs) }
+                        ) {
+                            Text(stringResource(R.string.debug_info_logs_view))
+                        }
+                    }
+                }
             }
         }
     }
 
-    fun shareArchive() {
-        model.generateZip()
+    private fun shareFile(file: File) {
+        val uri = FileProvider.getUriForFile(
+            this,
+            getString(R.string.authority_debug_provider),
+            file
+        )
+        val intent = Intent(Intent.ACTION_VIEW)
+        intent.setDataAndType(uri, "text/plain")
+        intent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION)
+        startActivity(Intent.createChooser(intent, null))
```

Comment:
Can we use `shareFile` for the zip file too? We would have to (optionally) pass subject, text, and type.

I think we should also always use `ShareCompat.IntentBuilder` for sharing files because it sets the read permission flag automatically (and the logic when to set which flag on which API level etc. is more complex than it may seem, see `ShareCompat` source code).